### PR TITLE
Add restart warning for auto complete settings

### LINKED ISSUE #6351: autocompletion always turned on; cannot be disabled
JabRef 5.1--2020-04-23--bbea7df
Linux 5.5.17-200.fc31.x86_64 amd64 
Java 14.0.1

I am using a custom field in the entry editor. Until recently this field was not assigned autocompletion for the phrases in the field. I did not turn on autocompletion for the field.

The bug consist of unwanted autocompletion that I cannot turn off.

### PR DESCRIPTION
**Pull request by ShiqingLiu**
To make auto complete settings come into effect, restart is needed. I add dialog box when someone change the auto complete settings, to remind him restart the software. 
Fixes #6351
screenshot

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java [hunk 1]
```diff
@@ -106,7 +109,13 @@ public void storeSettings() {
         } else if (firstNameModeFullProperty.getValue()) {
             firstNameMode = AutoCompleteFirstNameMode.ONLY_FULL;
         }
-
+        if(initialAutoCompletePreferences.shouldAutoComplete()!=enableAutoCompleteProperty.getValue()){
+            if(enableAutoCompleteProperty.getValue()){
+                restartWarnings.add(Localization.lang("Auto complete enable."));
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java:114]:
```suggestion
                restartWarnings.add(Localization.lang("Auto complete enabled."));
```

File: src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java [hunk 2]
```diff
@@ -106,7 +109,13 @@ public void storeSettings() {
...
+            }else{
+                restartWarnings.add(Localization.lang("Auto complete disable."));
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java:116]:
```suggestion
                restartWarnings.add(Localization.lang("Auto complete disabled."));
```

File: src/test/java/org/jabref/gui/preference/SaveAutoCompleteTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import java.lang.reflect.Field;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class SaveAutoCompleteTest {
+
+    @Test
+    public void storeAutoCompletePreferencesTest() throws NoSuchFieldException, IllegalAccessException {
+        JabRefPreferences preferencesService = JabRefPreferences.getInstance();
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/test/java/org/jabref/gui/preference/SaveAutoCompleteTest.java:19]: The test actually does not make any sense. You are just testing that a boolean variable is set in the preferences. This is useless and totally unrelated to your changes.
if you want to write a test, you need to test the View Model to check if your changed are correctly applied. But that's more complicated.

Manual testing should be enough. In jabref we focus on writing tests for important logic or other parts and not just writing tests for the sake of coverage.
So please remove your test.

[the remark this answers is not in the document]
Comment (ShiqingLiu) [reply src/test/java/org/jabref/gui/preference/SaveAutoCompleteTest.java:19]: Yeah, you are right. That test makes no sense for this issue. This was my early attempt while locating the code. It is really kind for you to point it out as well as two typos. Thanks. And I am still learning Github and its features.

File: src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java [hunk 4]
```diff
@@ -106,7 +109,13 @@ public void storeSettings() {
         } else if (firstNameModeFullProperty.getValue()) {
             firstNameMode = AutoCompleteFirstNameMode.ONLY_FULL;
         }
-
+        if(initialAutoCompletePreferences.shouldAutoComplete()!=enableAutoCompleteProperty.getValue()){
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java:112]: you have a lot of checkstyle issues here. Please setup the JabRef Code Style and make your life easier

Comment (ShiqingLiu) [reply src/main/java/org/jabref/gui/preferences/EntryEditorTabViewModel.java:112]: Thanks for your review.

**Comment by koppor** [comment]
This is a good start. Since there was no activity since a few weeks and there are minor things to fix to get this ready, we will take over here. Thank you for having been part at the JabRef development.

**Comment by calixtus** [comment]
I took the liberty to finish this one in another PR, your work is preserved an we will give you credit in the next release of JabRef. Thank you for your work!

**Pull request closed without merging**